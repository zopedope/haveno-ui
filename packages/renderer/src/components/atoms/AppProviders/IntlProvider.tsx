// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { useEffect, useMemo, useState } from "react";
import { IntlProvider as ReacIntlProvider } from "react-intl";
import type { FC } from "react";
import { LangPack, SupportedLocales } from "@src/constants/lang";
import type { LangKeys } from "@src/constants/lang";

const DEFAULT_LOCALE = SupportedLocales.EN;

export const IntlProvider: FC = ({ children }) => {
  const [messages, setMessages] = useState<{ [key in LangKeys]: string }>();
  const locale = useMemo(
    () =>
      navigator.language in SupportedLocales
        ? navigator.language
        : DEFAULT_LOCALE,
    [navigator.language]
  );

  useEffect(() => {
    setMessages(LangPack[locale] ?? LangPack[DEFAULT_LOCALE]);
  }, [locale]);

  return (
    <ReacIntlProvider locale={locale} messages={messages}>
      {children}
    </ReacIntlProvider>
  );
};
