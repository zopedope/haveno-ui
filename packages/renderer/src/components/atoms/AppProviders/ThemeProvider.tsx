// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { MantineProvider, Global } from "@mantine/core";
import type { FC } from "react";
import { themeOverride, globalStyles } from "@src/theme";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <MantineProvider theme={themeOverride}>
      <Global styles={globalStyles} />
      {children}
    </MantineProvider>
  );
};
