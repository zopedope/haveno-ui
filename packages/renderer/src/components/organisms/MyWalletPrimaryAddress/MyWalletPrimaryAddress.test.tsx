// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it, vi, beforeAll } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyWalletPrimaryAddress } from "./MyWalletPrimaryAddress";
import { AppProviders } from "@atoms/AppProviders";

describe("organisms::MyWalletPrimaryAddress", () => {
  beforeAll(() => {
    vi.mock("@hooks/haveno/useXmrPrimaryAddress", () => ({
      useXmrPrimaryAddress: () => ({
        isLoading: false,
        isSuccess: true,
        data: "HAVENOADDRESS",
      }),
    }));
  });

  it("renders without exploding", () => {
    const { asFragment, unmount } = render(
      <AppProviders>
        <MyWalletPrimaryAddress />
      </AppProviders>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it("renders xmr primary address from the daemon.", () => {
    const { unmount } = render(
      <AppProviders>
        <MyWalletPrimaryAddress />
      </AppProviders>
    );
    expect(screen.queryByText("HAVENOADDRESS")).toMatchSnapshot();
    unmount();
  });
});
