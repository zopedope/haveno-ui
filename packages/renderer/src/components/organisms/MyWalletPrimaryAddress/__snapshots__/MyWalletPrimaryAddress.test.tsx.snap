// Vitest Snapshot v1

exports[`organisms::MyWalletPrimaryAddress > renders without exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-Group-root mantine-sfnj26"
  >
    <div
      class="mantine-Stack-root mantine-DetailItem-root mantine-Group-child mantine-1a4fzbd"
    >
      <div
        class="mantine-Text-root mantine-DetailItem-label mantine-1ece26k"
      >
        Primary Address
      </div>
      <div
        class="mantine-Text-root mantine-DetailItem-content mantine-1tq817k"
      >
        <div
          class="mantine-Group-root mantine-1i932k4"
        >
          <div
            class="mantine-Group-child mantine-2w2ufd"
          >
            HAVENOADDRESS
          </div>
          <div
            class="mantine-Group-root mantine-Group-child mantine-ze53qg"
          >
            <a
              class="mantine-Text-root mantine-Anchor-root mantine-Group-child mantine-odvvvx"
            >
              Copy
            </a>
            <a
              class="mantine-Text-root mantine-Anchor-root mantine-Group-child mantine-odvvvx"
            >
              QR
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`organisms::MyWalletPrimaryAddress > renders xmr primary address from the daemon. 1`] = `
<div
  class="mantine-Group-child mantine-2w2ufd"
>
  HAVENOADDRESS
</div>
`;
